src/api/cpp/array.cpp

```diff
@@ -100,93 +135,71 @@ struct array::array_proxy::array_proxy_impl {
     void delete_on_destruction(bool val) { delete_on_destruction_ = val; }
 
     ~array_proxy_impl() {
-        if (delete_on_destruction_) delete parent_;
+        if (delete_on_destruction_) { delete parent_; }
     }
 
-   private:
-    array_proxy_impl(const array_proxy_impl &);
-    array_proxy_impl(const array_proxy_impl &&);
-    array_proxy_impl operator=(const array_proxy_impl &);
-    array_proxy_impl operator=(const array_proxy_impl &&);
+    array_proxy_impl(const array_proxy_impl &)  = delete;
+    array_proxy_impl(const array_proxy_impl &&) = delete;
+    array_proxy_impl operator=(const array_proxy_impl &) = delete;
+    array_proxy_impl operator=(const array_proxy_impl &&) = delete;
```

Comment:
We should keep refactoring & style changes separate from the change like this. Perhaps a separate commit where the array_proxy_impl copy and move constructors are deleted.

Reply:
This is equivalent to the previous code. Previously we made these constructors private which is the same as deleting them. Here we are using the `= delete` syntax.